python/tank/descriptor/io_descriptor/appstore.py

```diff
@@ -170,7 +170,8 @@ def __load_cached_app_store_metadata(self, path):
         :param path: path to bundle location on disk
         :return: metadata dictionary or None if not found
         """
-        cache_file = os.path.join(path, METADATA_FILE)
+        metadata_folder = self._get_metadata_folder(path)
```

Comment:
Won't this break backwards compatibility with currently cached bundles? It seems like we are. If that's indeed the case, a unit test proving it still works would be much appreciated. :)